backend/api/viewsets/CreditTrade.py

```diff
@@ -137,6 +137,8 @@ def list(self, request, *args, **kwargs):
         return response
 
     def perform_create(self, serializer):
+        agreement_date = self.request.data.get('agreement_date')  # Get the agreementDate from request data
```

Comment:
Why not just use date_of_written_agreement here? We wouldn't have to change it between agreement_date and date_of_written_agreement.

Reply:
You mean on frontend or agreement_date value we assigning in variable ?

Reply:
done